src/common/type/stringId.c

```diff
@@ -0,0 +1,375 @@
+
+    // Remove header to get the encoded characters
+    strId >>= STRING_ID_HEADER_SIZE;
+
+    // Decoding type
+    switch (bit)
+    {
+        // 5-bit decoding
+        case stringIdBit5:
+        {
+            // Map to convert encoding to characters
+            const char map[32] = "!abcdefghijklmnopqrstuvwxyz-1234";
+
+            // Macro to decode all but the last character
+            #define STR5ID_TO_ZN_IDX(idx)                                                                                          \
+                buffer[idx] = map[strId & 0x1F];                                                                                   \
+                strId >>= 5;                                                                                                       \
+                                                                                                                                   \
+                if (strId == 0)                                                                                                    \
+                    FUNCTION_TEST_RETURN(idx + 1)
+
+            // Char 1-11
+            STR5ID_TO_ZN_IDX(0);
+            STR5ID_TO_ZN_IDX(1);
+            STR5ID_TO_ZN_IDX(2);
+            STR5ID_TO_ZN_IDX(3);
+            STR5ID_TO_ZN_IDX(4);
+            STR5ID_TO_ZN_IDX(5);
+            STR5ID_TO_ZN_IDX(6);
+            STR5ID_TO_ZN_IDX(7);
+            STR5ID_TO_ZN_IDX(8);
+            STR5ID_TO_ZN_IDX(9);
+            STR5ID_TO_ZN_IDX(10);
+
+            // Char 12
+            buffer[11] = map[strId & 0x1F];
+            ASSERT(strId >> 5 == 0);
+
+            // If prefix flag is set then append +
+            if (prefix)
+            {
+                buffer[12] = '+';
+                FUNCTION_TEST_RETURN(13);
+            }
+
+            FUNCTION_TEST_RETURN(12);
+        }
+
+        // 6-bit decoding
+        default:
+        {
+            CHECK(bit == stringIdBit6);
+
+            // Map to convert encoding to characters
+            const char map[64] = "!abcdefghijklmnopqrstuvwxyz-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+            // Macro to decode all but the last character
+            #define STR6ID_TO_ZN_IDX(idx)                                                                                          \
+                buffer[idx] = map[strId & 0x3F];                                                                                   \
+                strId >>= 6;                                                                                                       \
+                                                                                                                                   \
+                if (strId == 0)                                                                                                    \
+                    FUNCTION_TEST_RETURN(idx + 1)
+
+            // Char 1-9
+            STR6ID_TO_ZN_IDX(0);
```

Comment:
Can't we do this in a loop and use the STRID6_MAX like 
```
for (unsigned int idx = 0, idx < STRID6_MAX - 3; idx++)
    STR6ID_TO_ZN_IDX(idx);
```

Reply:
Yes, but this is much faster.